From dc113bd6d935140178fe81642efee05574265bb0 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:42:33 -0400
Subject: [ethernet] cxgb4: Stop using ethtool SPEED_* constants

Message-id: <1410475447-94392-15-git-send-email-svemuri@redhat.com>
Patchwork-id: 93792
O-Subject: [RHEL7.1 PATCH BZ 1124947 014/108] cgxb4: Stop using ethtool SPEED_* constants
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

ethtool speed values are just numbers of megabits and there is no need
to add SPEED_40000.  To be consistent, use integer constants directly
for all speeds.

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit e8b39015b591f359d464d026170dcf76f7ed5b8e)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
index 0e7ae06..70b38d9 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
@@ -427,16 +427,16 @@ static void link_report(struct net_device *dev)
   const struct port_info *p = netdev_priv(dev);
 
   switch (p->link_cfg.speed) {
-  case SPEED_10000:
+  case 10000:
    s = "10Gbps";
    break;
-  case SPEED_1000:
+  case 1000:
    s = "1000Mbps";
    break;
-  case SPEED_100:
+  case 100:
    s = "100Mbps";
    break;
-  case 40000: /* Need a SPEED_40000 in ethtool.h */
+  case 40000:
    s = "40Gbps";
    break;
   }
@@ -2274,13 +2274,13 @@ static int get_settings(struct net_device *dev, struct ethtool_cmd *cmd)
 
 static unsigned int speed_to_caps(int speed)
 {
- if (speed == SPEED_100)
+ if (speed == 100)
   return FW_PORT_CAP_SPEED_100M;
- if (speed == SPEED_1000)
+ if (speed == 1000)
   return FW_PORT_CAP_SPEED_1G;
- if (speed == SPEED_10000)
+ if (speed == 10000)
   return FW_PORT_CAP_SPEED_10G;
- if (speed == 40000) /* Need SPEED_40000 in ethtool.h */
+ if (speed == 40000)
   return FW_PORT_CAP_SPEED_40G;
  return 0;
 }
@@ -2310,8 +2310,8 @@ static int set_settings(struct net_device *dev, struct ethtool_cmd *cmd)
   cap = speed_to_caps(speed);
 
   if (!(lc->supported & cap) ||
-      (speed == SPEED_1000) ||
-      (speed == SPEED_10000) ||
+      (speed == 1000) ||
+      (speed == 10000) ||
       (speed == 40000))
    return -EINVAL;
   lc->requested_speed = cap;
diff --git a/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c b/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
index 13d7690..2428ca2 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
@@ -3568,13 +3568,13 @@ int t4_handle_fw_rpl(struct adapter *adap, const __be64 *rpl)
   if (stat & FW_PORT_CMD_TXPAUSE)
    fc |= PAUSE_TX;
   if (stat & FW_PORT_CMD_LSPEED(FW_PORT_CAP_SPEED_100M))
-   speed = SPEED_100;
+   speed = 100;
   else if (stat & FW_PORT_CMD_LSPEED(FW_PORT_CAP_SPEED_1G))
-   speed = SPEED_1000;
+   speed = 1000;
   else if (stat & FW_PORT_CMD_LSPEED(FW_PORT_CAP_SPEED_10G))
-   speed = SPEED_10000;
+   speed = 10000;
   else if (stat & FW_PORT_CMD_LSPEED(FW_PORT_CAP_SPEED_40G))
-   speed = 40000; /* Need SPEED_40000 in ethtool.h */
+   speed = 40000;
 
   if (link_ok != lc->link_ok || speed != lc->speed ||
       fc != lc->fc) {                    /* something changed */
-- 
1.7.1